From fbe51e7d6cf3e2c5110d07aa825869883400da75 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.j.ledkov@intel.com>
Date: Thu, 14 May 2015 12:36:41 +0100
Subject: [PATCH 6/7] Use our ld.so, and our libraries. lsof output will be
 interesting.
Organization: Intel Corporation (UK) Ltd. - Co. Reg. #1134945 - Pipers Way, Swindon SN3 1RJ

---
 stage1/init/init.go | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/stage1/init/init.go b/stage1/init/init.go
index df353a5..3f453ea 100644
--- a/stage1/init/init.go
+++ b/stage1/init/init.go
@@ -61,7 +61,7 @@ const (
 	lkvmBin = "/usr/bin/lkvm"
 	bzImg = "/usr/lib/kernel/vmlinux.container"
 	// Path to the interpreter within the stage1 rootfs
-	interpBin = "/usr/lib/ld-linux-x86-64.so.2"
+	interpBin = "/usr/lib64/ld-linux-x86-64.so.2"
 	// Path to the localtime file/symlink in host
 	localtimePath = "/etc/localtime"
 )
@@ -128,6 +128,11 @@ func getArgsEnvNspawn(p *Pod) ([]string, []string, error) {
 	args := []string{}
 	env := os.Environ()
 
+	args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), interpBin))
+	args = append(args, "--library-path")
+	args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), "usr/lib64"))
+
+
 	args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), nspawnBin))
 	args = append(args, "--boot") // Launch systemd in the pod
 	out, err := os.Getwd()
@@ -177,6 +182,10 @@ func getArgsEnvKvm(p *Pod) ([]string, []string, error) {
 	kargs := []string{}
 	env := os.Environ()
 
+	args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), interpBin))
+	args = append(args, "--library-path")
+	args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), "usr/lib64"))
+
 	args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), lkvmBin))
 	args = append(args, "run")
 
-- 
2.1.4

